hagrid/reservations/export_pdf.py

```diff
@@ -0,0 +1,342 @@
+    render_collection_table_header(d)
+    for request in l:
+        if d.cursor_y < (75 + 15):
+            d.new_page()
+            render_collection_table_header(d)
+        render_collection_list_entry(request[0], request[1], d)
+
+
+def render_settlement_head(d: Document):
+    d.canvas.setFont("Helvetica", 11)
+    d.cursor_y -= 30
+    if d.cursor_y < (75 + 15):
+        d.new_page()
+    d.canvas.drawString(d.cursor_x, d.cursor_y - 5, "settlement: ")
+    d.cursor_y -= 20
+
+    # Render table head
+    p = d.canvas
+    p.line(45, d.cursor_y, 45, d.cursor_y - 15)
+    p.line(d.w - 45, d.cursor_y, d.w - 45, d.cursor_y - 15)
+    p.line(45, d.cursor_y, d.w - 45, d.cursor_y)
+    p.line(45, d.cursor_y - 15, d.w - 45, d.cursor_y - 15)
+    p.drawString(50, d.cursor_y - 10, "Amount")
+    p.drawString(100, d.cursor_y - 10, "Article")
+    p.drawString(250, d.cursor_y - 10, "Single Item price")
+    p.drawString(d.w - 175, d.cursor_y - 10, "Sum")
+    d.cursor_y -= 15
+    d.canvas.setFont("Helvetica", 9)
+
+
+def render_invoice_end(l, d: Document):
+    render_settlement_head(d)
+    total = 0
+    for request in l:
+        if d.cursor_y < (75 + 15):
+            d.new_page()
+        amount: int = request[1]
+        a: Variation = request[0]
+        price = a.product.price
+        total += price * amount
+        d.canvas.line(45, d.cursor_y, 45, d.cursor_y - 15)
+        d.canvas.line(d.w - 45, d.cursor_y, d.w - 45, d.cursor_y - 15)
+        d.canvas.drawString(100, d.cursor_y - 10, str(a))
+        d.canvas.drawString(50, d.cursor_y - 10, str(amount))
+        d.canvas.drawString(250, d.cursor_y - 10, '{:20,.2f} €'.format(price))
+        d.canvas.drawString(d.w - 175, d.cursor_y - 10, '{:20,.2f} €'.format(price * amount))
+        d.cursor_y -= 15
+    cy = d.cursor_y
+    d.canvas.line(45, cy, d.w - 45, cy)
+    d.canvas.line(45, cy, 45, cy - 15)
+    d.canvas.line(d.w - 45, cy, d.w - 45, cy - 15)
+    d.canvas.line(45, cy - 15, d.w - 45, cy - 15)
+    d.canvas.drawString(50, cy - 10, "TOTAL:")
+    d.canvas.drawString(100, cy - 10, "[EUR]")
+    d.canvas.drawString(d.w - 175, cy - 10, '{:20,.2f} €'.format(total))
+    # Signature of person who packed and person who checked
+    d.cursor_y -= 75
+    d.canvas.line(55, d.cursor_y, 235, d.cursor_y)
+    d.canvas.drawString(60, d.cursor_y - 10, "Signature of person who packed")
+    d.canvas.line(255, d.cursor_y, 435, d.cursor_y)
+    d.canvas.drawString(260, d.cursor_y - 10, "Signature of person who checked")
+
+
+def render_reservation(r: Reservation, d: Document):
+    if r.state == r.STATE_SUBMITTED:
+        d.set_watermark("")
+    else:
+        d.set_watermark(str(r.state))
+    articles = generate_collection_list(r)
+    render_invoice_header(r, d)
+    render_collection_list(articles, d)
+    render_invoice_end(articles, d)
+
+
+def export_invoices_as_pdf(reservations, filename: str, username = "nobody", title = "C3FOC - Reservations"):
```

Comment:
I think we should not call it an invoice, as it is really not an invoice. Something like `generate_reservation_pdf` might be a better fit.

Reply:
I agree on the fact that "invoice" is a bad name. But I also think that 

    generate_reservation_pdf

is equally bad since there may be multiple printable versions of the same group reservation (e.g. one that displays each persons individual reservation for reference inside the teams).

Reply:
`generate_packing_pdf` might be a good name.